test/expectations.go

```diff
@@ -0,0 +1,73 @@
+package test
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"time"
+
+	"github.com/awslabs/operatorpkg/object"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+const (
+	SlowTimeout = 100 * time.Second
+	SlowPolling = 10 * time.Second
+	FastTimeout = 1 * time.Second
+	FastPolling = 10 * time.Millisecond
+)
+
+func ExpectReconcile(ctx context.Context, controller reconcile.Reconciler, object client.Object) reconcile.Result {
```

Comment:
Given that we almost always expect it to succeed, I'm thinking of making it `ExpectReconciled`, and we can skip having a helper for the failure case, since it's so rare. You can do `Expect(controller.Reconcile(nn)).ToNot(Succeed())`